src/Nethermind/Nethermind.Logging/PathUtils.cs

```diff
@@ -25,7 +28,17 @@ public static class PathUtils
     {
         public static string GetExecutingDirectory()
         {
-            return Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);;
+            var process = Process.GetCurrentProcess();
+            if (process.ProcessName.StartsWith("dotnet", StringComparison.InvariantCultureIgnoreCase))
+            {
+                return Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+            }
+            
+            var filePath = process.MainModule.FileName;
+            var fileName = filePath.Split(Path.DirectorySeparatorChar).LastOrDefault();
+            var fileNameIndex = filePath.LastIndexOf(fileName, StringComparison.InvariantCultureIgnoreCase);
+
+            return filePath.Substring(0, fileNameIndex);
         }
```

Comment:
Having a method like:

        public static string GetApplicationResourcePath(this string resourcePath, string overridePrefixPath = null)
        {
            if (string.IsNullOrEmpty(overridePrefixPath))
            {
                return Path.Combine(ExecutingDirectory, resourcePath);
            }
            else if (Path.IsPathRooted(overridePrefixPath))
            {
                return Path.Combine(overridePrefixPath, resourcePath);
            }
            else
            {
                return Path.Combine(ExecutingDirectory, overridePrefixPath, resourcePath);
            }
        }

would probably cover most of our use cases and we wouldn't have to leak path abstraction logic everywhere.